shapely/plotting.py

```diff
@@ -5,28 +5,53 @@
 exploration, debugging and illustration purposes.
 
 """
+
+import matplotlib.pyplot as plt
 import numpy as np
+from matplotlib import colors
+from matplotlib.patches import PathPatch
+from matplotlib.path import Path
```

Comment:
these top-level imports will need to be nested to the functions below, since matplotlib is not a required dependency